tests/test_google_analytics_start_date.py

```diff
@@ -0,0 +1,291 @@
+
+    def expected_default_fields(self):
+        return {
+            "report 1" : {},
+            "Audience Overview": {"ga:users","ga:newUsers","ga:sessions","ga:sessionsPerUser","ga:pageviews","ga:pageviewsPerSession","ga:avgSessionDuration","ga:bounceRate","ga:date"},
+            "Audience Geo Location": {"ga:users","ga:newUsers","ga:sessions","ga:pageviewsPerSession","ga:avgSessionDuration","ga:bounceRate","ga:date","ga:country","ga:city","ga:continent","ga:subContinent"},
+            "Audience Technology": {"ga:users","ga:newUsers","ga:sessions","ga:pageviewsPerSession","ga:avgSessionDuration","ga:bounceRate","ga:date","ga:browser","ga:operatingSystem"},
+            "Acquisition Overview": {"ga:sessions","ga:pageviewsPerSession","ga:avgSessionDuration","ga:bounceRate","ga:acquisitionTrafficChannel","ga:acquisitionSource","ga:acquisitionSourceMedium","ga:acquisitionMedium"},
+            "Behavior Overview": {"ga:pageviews","ga:uniquePageviews","ga:avgTimeOnPage","ga:bounceRate","ga:exitRate","ga:exits","ga:date","ga:pagePath","ga:pageTitle"},
+            "Ecommerce Overview": {"ga:transactions","ga:transactionId","ga:campaign","ga:source","ga:medium","ga:keyword","ga:socialNetwork"}
+        }
+
+    def get_properties(self, original: bool = True):
+        return_value = {
+            'start_date' : dt.strftime(dt.utcnow() - timedelta(days=30), self.START_DATE_FORMAT),
+            'view_id': os.getenv('TAP_GOOGLE_ANALYTICS_VIEW_ID'),
+            'report_definitions': [{"id": "a665732c-d18b-445c-89b2-5ca8928a7305", "name": "report 1"}]
+        }
+        if original:
+            return return_value
+        
+        # Start Date test needs the new connections start date to be prior to the default
+        assert self.START_DATE < return_value["start_date"]
+
+        # Assign start date to be the default 
+        return_value["start_date"] = self.START_DATE
+        return return_value
+
+    def get_field_selection(self):
+        return set()
+
+    def select_all_catalogs(self, conn_id, found_catalogs):
+        # select all catalogs
+        for c in found_catalogs:
+            catalog_entry = menagerie.get_annotated_schema(conn_id, c['stream_id'])
+
+            for k in self.expected_automatic_fields()[c['stream_name']]:
+                mdata = next((m for m in catalog_entry['metadata']
+                              if len(m['breadcrumb']) == 2 and m['breadcrumb'][1] == k), None)
+                print("Validating inclusion on {}: {}".format(c['stream_name'], mdata))
+                self.assertTrue(mdata and mdata['metadata']['inclusion'] == 'automatic')
+
+            keys = set(catalog_entry['annotated-schema']['properties'].keys())
+            non_selected_fields = keys - self.get_field_selection()
+            connections.select_catalog_via_metadata(conn_id,
+                                                    c,
+                                                    catalog_entry)
+
+    def test_run(self):
```

Comment:
There's a bunch of repeated code in the test run. Could you take a stab at refactoring it to consolidate some into a base class? So that each test only has unique test code?